python/ray/autoscaler/updater.py

```diff
@@ -304,7 +303,7 @@ def ssh_cmd(self,
         except subprocess.CalledProcessError:
             logger.error("Command failed: \n\n  {}\n".format(
                 " ".join(final_cmd)))
-            sys.exit(1)
```

Comment:
Maybe this could be `raise SystemExit(1)` and then we catch it on the upper level? Or does `raise SystemExit(1)` behave in exactly the same way?

Alternatively, we might want to use the `expect_error` argument to determine whether to throw or exit. Not sure if that's the best of handling this though.

Reply:
Hm, maybe expect_error should be replaced with `raise_on_error`, and get rid of `expect_error`? Otherwise, `expect_error=True` will always be silent, and we can't toggle raising.